### ISSUE TITLE
All exceptions are caught

### ISSUE THREAD

**Issue by rovansteen** [Youshido/GraphQL#97]
Why are all the exceptions caught? This makes debugging really difficult, because there will be no logs or for example in my Symfony application nothing shows up in the profiler. Am I missing something here?

**Comment by viniychuk** [comment]
@rovansteen Will let it slip in Symfony bundle. Ideally – exceptions will be delivered with the descriptive information so you can easily see what's the problem is. That was a big topic of discussion and idea was to return response in any case... but again, we have to keep the original exception message if it's generic.

**Comment by rovansteen** [comment]
@viniychuk yes I understand that. Perhaps having some sort of middleware where you can do something with the exception, like log it or send it to a service like Sentry will help with debugging a lot while still being able to send a useful response back to the client.

**Comment by garte** [comment]
Just a workaround: I've written my own ExecutionContext class and invoke it via the given parameter "graphql.execution_context.class". This way I can override the "addError" method and log accordingly.

**Comment by lox** [comment]
This issue makes testing and debugging really difficult. It seems like there are lots of places where `\Exception` is caught. Perhaps a pluggable exception handler would be a good way of dealing with the problem?
